doc/cookbook/named-routes/NamedRoutes.lhs

```diff
@@ -0,0 +1,390 @@
+  parseQueryParam param = Left $ param <> " is not a valid SortBy"
+```
+
+So, remember, this type represents the `MoviesAPI` node that we’ve connected earlier to the main `API` tree.
+
+In this subtree, we illustrated both an endpoint with a **query param** and also, a **capture** with a subtree underneath it.
+
+The first branch is done, now, let's also implement the second one as follows: 
+
+```haskell
+data MovieAPI mode = MovieAPI
+  { get :: mode :- Get '[JSON] (Maybe Movie)
+  , update :: mode :- ReqBody '[JSON] Movie :> Put '[JSON] NoContent
+  , delete :: mode :- Delete '[JSON] NoContent
+  } deriving stock Generic
+```
+
+Small detail: as our main API tree is also a record, we need the `NamedRoutes` helper (to obtain the API proper)
+To improve readability, we suggest you create a type alias:
+
+```haskell
+type MovieCatalogAPI = NamedRoutes API
+```
+
+That's it, we have our `MovieCatalogAPI` type!
+
+Let's make a server and a client out of it!
+
+## The Server
+
+As you know, we can’t talk about a server, without addressing the handlers.
+
+First, we build our handlers (mind that we're cheating a bit, obviously, the `moviesDB` in reality 
+would have to be part of some state, such that we can modify it, additionally, making it a list is not 
+very wise in terms of performance)
+
+```haskell
+versionHandler :: Handler Version
+versionHandler = pure "0.0.1"
+
+movieListHandler :: Maybe SortBy -> Handler [Movie]
+movieListHandler _ = 
+  -- depending on sortBy, do a different sorting
+  pure moviesDB
+
+moviesDB :: [Movie]
+moviesDB =
+  [ Movie "1" "Se7en" 1995
+  , Movie "2" "Minority Report" 2002
+  , Movie "3" "The Godfather" 1972
+  ]
+
+getMovieHandler :: MovieId -> Handler (Maybe Movie)
+getMovieHandler requestMovieId = pure $ find (\movie -> movie.movieId == requestMovieId) moviesDB
+
+updateMovieHandler :: MovieId -> Movie -> Handler NoContent
+updateMovieHandler requestedMovieId newMovie =
+   -- update the movie list in the database...
+   pure NoContent
+
+deleteMovieHandler :: MovieId -> Handler NoContent
+deleteMovieHandler _ =
+   -- delete the movie from the database...
+   pure NoContent
+```
+
+And assemble them together with the record structure, which is the glue here.
+
+```haskell
+server ::  API AsServer
+server =
+  API
+    { version = versionHandler
+    , movies = moviesHandler
+    }
+
+moviesHandler :: MoviesAPI AsServer
+moviesHandler =
+  MoviesAPI
+    { list = movieListHandler
+    , movie = movieHandler
+    }
+
+movieHandler :: MovieId -> MovieAPI AsServer
+movieHandler mId = MovieAPI
+    { get = getMovieHandler mId
+    , update = updateMovieHandler mId
+    , delete = deleteMovieHandler mId
+    }
+```
+
+Finally, we can run the server and connect the API routes to the handlers, using the convenient `genericServe` function
+
+```haskell
+main :: IO ()
+main = run 8081 app
+
+app :: Application
+app = genericServe server
+```
+
+Yay! That’s done and we’ve got our server!
```

Comment:
Seems like these are `U+2019 RIGHT SINGLE QUOTATION MARK`. I'd prefer to consistently use `U+0027 APOSTROPHE`.

Reply:
I'd love to hear if there is a specific reason why they're unicode quotes, I don't know much about `lhs`.

Reply:
Probably written by the original author, I don’t think I usually use these on my laptop keyboard

Reply:
Will adjust